pywbem/_logging.py

```diff
@@ -0,0 +1,402 @@
+      separate pywbem logging attributes.
+
+    """
+    __metaclass__ = MetaPywbemLoggers
+
+    @classmethod
+    def reset(cls):
+        """Reset the logger dictionary. Used primarily in unittests"""
+        cls.loggers = {}
+
+    @classmethod
+    def create_loggers(cls, input_str, log_filename=None):
+        """
+        Create the pywbem loggers defined by the input string in the following
+        format and place in a class level dictionary in this class:
+
+        Parameters:
+          input_str (:term:`string`) that specifies the logger definitions:
+            as follows:
+
+            log_spec[,log_spec]
+
+            log_spec := log_comp['=' [dest][":"[detail_level][":"[log_level]]]]
+
+            where:
+                comp (str) must be one of PYWBEM_LOG_COMPONENTS
+                log_level (str) must be one of LOG_LEVELS above
+                detail_level (str) must be one of LOG_DETAIL_LEVELS
+                dest (str) must be one of LOG_DESTINATIONS
+
+          log_filename (:term:`string`)
+            Optional string that defines the filename for output of logs
+            if the dest type is `file`
+
+        Exceptions:
+          ValueError if any of the parameters are invalid
+        """
+        results = cls._parse_log_specs(input_str)
+        for name, value in six.iteritems(results):
+            cls.create_logger(name, log_dest=value[0],
+                              log_detail_level=value[1],
+                              log_level=value[2],
+                              log_filename=log_filename)
+
+    @classmethod
+    def create_logger(cls, log_component, log_dest='stderr',
+                      log_filename=None,
+                      log_detail_level='all', log_level='debug'):
+        """
+        Setup a single named logger with the characteristics defined on input.
+
+        This function can be used to set up all of the named loggers used by
+        pywbem.
+
+        Parameters:
+          log_compnent (:term:`string`):
+           The name of the logger. It must be one of the
+           names defined in PYWBEM_LOG_COMPONENTS. Subsequent logging users
+           will use this name to reference the parameters of this logger
+
+          log_dest (:term:`string`):
+            String defining the destination for this log. It must be one of the
+            destinations defined in LOG_DESTINATIONS
+
+          log_filename (:term:`string`):
+            Filename to use as logging file if the log destination is `file`.
+            Ignored if log destination is not `file`. The default is
+            DEFAULT_LOG_FILENAME defined in config.py
+
+          log_detail_level (:term:`string`):
+            String defining the level of detail for log output. This is
+            optional.
+            Not all of the loggers use this attribute.  Thus, the ops and http
+            loggers used a defined detail level and always set logging to that
+            level.
```

Comment:
DISCUSSION:  right now we fix each logger level.  I think we should use this as the level at which we log.  That is only documentation change here but I will have to change the recorder to get the log level from here rather than fixing it.